src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/EmbeddingPreview.jsx

```diff
@@ -17,487 +16,261 @@ import PlotStyling from '../../filter-cells/components/PlotStyling';
-          titleFontSize: { value: config.axisTitlesize },
-          labelFontSize: { value: config.axisTicks },
-          offset: { value: config.axisOffset },
-          gridOpacity: { value: (config.transGrid / 20) },
-        },
-        {
-          scale: 'y',
-          grid: true,
-          domain: true,
-          orient: 'left',
-          titlePadding: 5,
-          title: { value: config.yAxisText2 },
-          titleFont: { value: config.masterFont },
-          labelFont: { value: config.masterFont },
-          titleFontSize: { value: config.axisTitlesize },
-          labelFontSize: { value: config.axisTicks },
-          offset: { value: config.axisOffset },
-          gridOpacity: { value: (config.transGrid / 20) },
-        },
-      ],
-      marks: [
-        {
-          type: 'symbol',
-          from: { data: 'embeddingCat' },
-          encode: {
-            enter: {
-              x: { scale: 'x', field: 'umap1' },
-              y: { scale: 'y', field: 'umap2' },
-              size: { value: config.pointSize },
-              stroke: {
-                scale: 'color',
-                field: 'geneExpression',
-              },
-              fill: {
-                scale: 'color',
-                field: 'geneExpression',
-              },
-              shape: { value: config.pointStyle },
-              fillOpacity: { value: config.pointOpa / 10 },
-            },
-          },
-        },
-
-      ],
-      legends: config.legend,
-      title:
-      {
-        text: { value: config.titleText },
-        color: { value: config.masterColour },
-        anchor: { value: config.titleAnchor },
-        font: { value: config.masterFont },
-        dx: { value: config.bounceX },
-        fontSize: { value: config.titleSize },
-      },
-    };
-  }
-
-  generateData() {
-    const { data } = this.state;
-
-    return data;
-  }
-
-  updatePlotWithChanges(obj) {
-    this.setState((prevState) => {
-      const newState = _.cloneDeep(prevState);
-
-      _.merge(newState.config, obj);
-
-      return newState;
+    const spec = plots[selectedSpec].specGenerator(config);
+    generateData(spec);
+    console.log(config, spec);
+    setPlotSpec(spec);
+  }, [config]);
+
+  useEffect(() => {
+    setConfig(plots[selectedSpec].initialConfig);
+  }, [selectedSpec]);
+
+  // Quick and dirty function to massage prepared data into a good shape.
+  // This will be changed once we actually load data from Redux.
+  const generateData = (spec) => {
+    spec.data.map((s) => {
+      if (s.name === 'cellSets') {
+        s.values = [];
+
+        UMAP.forEach((cell, i) => {
+          s.values[cell.cluster_id] = {
+            name: `${cell.cluster_id}`,
+            cellSetId: cell.cluster_id,
+            cellIds: s.values[cell.cluster_id]?.cellIds ? [...s.values[cell.cluster_id].cellIds, i] : [i],
+            color: colorProvider.getColor(),
+          };
+          // cluster_id
```

Comment:
this comment does not look very useful, you can remove it

Reply:
Fixed.